utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -353,15 +384,33 @@ trait IntervalArray[K <: Interval[K], T] extends Serializable {
     //    of the query region, and thus, there cannot be any intervals earlier
     //    in the array that overlap the query region.
 
-    val optIdx = binarySearch(rr)
+    optLastIndex = optLastIndex.filter(array(_)._1.covers(rr)).orElse(binarySearch(rr))
 
-    optIdx.toIterable
+    optLastIndex.toIterable
       .flatMap(idx => {
         expandBackward(rr, idx) ::: expandForward(rr, idx + 1)
       })
   }
 
   /**
+   * Gets all overlapping regions for a given K, however if no overlapping
+   * regions are found, uses the element that is the closest to the query.
+   *
+   * @see get
+   *
+   * @param rr interval to filter by.
+   * @return iterable of elements filtered by region rr containing overlapping
+   *         or the closest element in array.
+   */
+  def getNearest(rr: K): Iterable[(K, T)] = {
```

Comment:
is there a literature ref for this algorithm, or other API examples?  `nearest` doesn't seem like the right word to me, though I can be convinced it is.